src/python/pants/engine/target.py

```diff
@@ -1592,9 +1592,13 @@ def __init__(
         expected_type: str,
         description_of_origin: str | None = None,
     ) -> None:
+        if hasattr(raw_value, "_type_description"):
+            raw_type = f"which is an {raw_value._type_description}"  # type: ignore[union-attr]
+        else:
+            raw_type = f"with type `{type(raw_value).__name__}`"
```

Comment:
This feels awkward:
- we assume "an" and not "a"
- there's this special magic attribute

Any way we can leverage `__repr__` or `__str__` on the object or type? Or even just override `__name__` on the type (if that doesn't break any assumptions made by CPython)?

Reply:
Honestly rewiring the `__name__` of an internal type feels less bad.
Something like `<unrecognized symbol>`.

Unless `str` on a class gives the unqualified name (e.g. `<class Foo>` and not `<class pants.whatever.Foo>` then we can just override the class's `__syr__`.

Reply:
Yea, I agree and did consider overriding `__name__` but that's seems worse to me.
My first attempt to get rid of this magic attribute was to drop the private `_` prefix on the class name and just import it and have a `if isinstance(raw_type, UnrecognizedSymbol)` check, but that led to circular imports. I could of course move the declaration out of `parser.py` to solve that, and perhaps that's the cleaner approach to take here..?

But I felt reluctant to do that, merely to fix the phrasing of an error message.

Also considered overriding the `__repr__` for this. (don't follow the unqualified concern, as the current use of just the types `__name__` is also unqualified, right?)

We could address the assumption of the `a` vs `an` on the magic attribute by shifting that part of the phrasing onto the attribute itself: `f"which is {raw_value._type_description}"

I'm honestly up for whichever way to address this really.. whatever your preferences tilt towards.